### PR TITLE
extra windows deps

### PR DESCRIPTION
**Pull request by gdams**
closes: https://github.com/AdoptOpenJDK/openjdk-infrastructure/issues/294

### CODE REVIEW

File: ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Make-3.82/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  win_get_url:
+    url: https://ci.adoptopenjdk.net/userContent/make-3.82.zip
+    dest: 'C:\temp\make-3.82.zip'
+  tags: Make-3.82
+
+- name: Extract make-3.82 to C:\openjdk\
+  win_unzip:
+    src: 'C:\temp\make-3.82.zip'
+    dest: 'C:\openjdk\'
+  tags: Make-3.82
```
Comment (gdams) [inline ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Make-3.82/tasks/main.yml]: I need to `./configure` and `make` here

File: ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Certificate/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+---
+##########################
+# Windows Code Sign Cert #
+##########################
+- name: Download Windows.p12
+  win_get_url:
+    url: https://github.com/AdoptOpenJDK/openjdk-installer/raw/master/windows.p12
+    dest: 'C:\Users\jenkins\windows.p12'
```
Comment (jdekonin) [inline ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Certificate/tasks/main.yml]: It would be good to document where this comes from originally.

File: ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Common/tasks/main.yml [hunk 3]
```diff
@@ -13,10 +13,15 @@
...
 - name: Create cmd.exe shortcut
   win_shortcut:
     src: C:\Windows\System32\cmd.exe
     dest: C:\Users\Public\Desktop\cmd.lnk
     icon: C:\Windows\System32\cmd.exe,0
   tags: basic_config
+
+- name: Set Hostname
+  raw: netdom renamecomputer "$env:COMPUTERNAME" /Newname "{{ inventory_hostname }}" -force
+  when: Domain == "adoptopenjdk.net"
```
Comment (jdekonin) [inline ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Common/tasks/main.yml]: Would this step not be wanted if the domain wasn't adoptopenjdk.net?  Example, a developer running this against their own cloud instance.

File: ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Freetype/tasks/main.yml [hunk 4]
```diff
@@ -39,3 +39,44 @@
...
+
+- name: Test if freetype is already downloaded
+  win_stat:
+    path: 'C:\temp\freetype.zip'
+  register: freetype_compiled_download
+  tags: Freetype
+
+- name: Download pre compiled freetype.zip
+  win_get_url:
+    url: https://ci.adoptopenjdk.net/userContent/freetype.zip
```
Comment (jdekonin) [inline ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Freetype/tasks/main.yml]: It would be good to document where this comes from originally.

File: ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/MS_SDK_7.1/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+- name: Test if MS SDK is already downloaded
+  win_stat:
+    path: 'C:\temp\MS_SDK.zip'
+  register: ms_sdk_download
+  tags: MS_SDK_7.1
+
+- name: Download pre compiled freetype.zip
+  win_get_url:
+    url: https://ci.adoptopenjdk.net/userContent/MicrosoftSDKs.zip
```
Comment (jdekonin) [inline ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/MS_SDK_7.1/tasks/main.yml]: It would be good to document where this comes from originally.

File: ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Make-3.82/tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+####################
+- name: Test if Make-3.82 is already installed
+  win_stat:
+    path: 'C:\openjdk\make-3.82'
+  register: make_382_installed
+  tags: Make-3.82
+
+- name: Download make-3.82
+  win_get_url:
+    url: 'https://ci.adoptopenjdk.net/userContent/make-3.82.zip'
```
Comment (jdekonin) [inline ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Make-3.82/tasks/main.yml]: It would be good to document where this comes from originally.

File: ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Common/tasks/main.yml (hunk 3 above)
Comment (gdams) [reply ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Common/tasks/main.yml]: I have just followed what @bblondin did on the unix playbook

Comment (jdekonin) [reply ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Common/tasks/main.yml]: @bblondin change has a case for Domain == "adoptopenjdk.net" and Domain != "adoptopenjdk.net"

File: ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/MS_SDK_7.1/tasks/main.yml (hunk 5 above)
Comment (sxa555) [reply ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/MS_SDK_7.1/tasks/main.yml]: Comment needs to be changed as it says freetype :-)

File: ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/cygwin/tasks/main.yml [hunk 7]
```diff
@@ -35,6 +35,7 @@
   win_command: C:\temp\cygwin.exe -q -P "{{ item }}"
   with_items:
     - autoconf
+    - gcc-core
```
Comment (sxa555) [inline ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/cygwin/tasks/main.yml]: I presume this role is still to be added? Doesn't seem to be part of this PR

File: ansible/playbooks/AdoptOpenJDK_Windows_Playbook/main.yml [hunk 8]
```diff
@@ -34,12 +34,15 @@
     - Freemarker
     - Freetype
     - cmake
+    - Make-3.82
```
Comment (sxa555) [inline ansible/playbooks/AdoptOpenJDK_Windows_Playbook/main.yml]: Is the make installed in the cygwin role not adequate?

Comment (gdams) [reply ansible/playbooks/AdoptOpenJDK_Windows_Playbook/main.yml]: to build openjdk8 it needs an older version of Make (not make 4)

File: ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/cygwin/tasks/main.yml (hunk 7 above)
Comment (gdams) [reply ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/cygwin/tasks/main.yml]: this just installs `gcc-core` into cygwin. It's not a role :)

File: ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Certificate/tasks/main.yml (hunk 2 above)
Comment (gdams) [reply ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Certificate/tasks/main.yml]: this is fetched from our installer repo where the .p12 file lives

Comment (tellison) [reply ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Certificate/tasks/main.yml]: I expect @jdekonin means document where the file came from originally before it was put into the installer repo.

I agree, though that's best documented in the installer repo alongside the file itself rather than here where the file is used.

File: ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Freetype/tasks/main.yml (hunk 4 above)
Comment (tellison) [reply ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Freetype/tasks/main.yml]: Right, but we should track where the *original* source came from for freetype, which version was used, how it was compiled, etc.  That's best done in the repo housing the zip rather than here where the zip is used.

File: ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Certificate/tasks/main.yml (hunk 2 above)
Comment (gdams) [reply ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Certificate/tasks/main.yml]: umm well it's our custom code signing certificate. Some of those details are in the secrets repo :)

File: ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/MS_SDK_7.1/tasks/main.yml [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+  register: ms_sdk_download
+  tags: MS_SDK_7.1
+
+- name: Download MicrosoftSDKs.zip
+  win_get_url:
+    url: https://ci.adoptopenjdk.net/userContent/MicrosoftSDKs.zip
+    dest: C:\temp\MS_SDK.zip
+  when: (ms_sdk_installed.stat.exists == false) and (ms_sdk_download.stat.exists == false)
+  tags: MS_SDK_7.1
+
```
Comment (sxa555) [inline ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/MS_SDK_7.1/tasks/main.yml]: I'm still a touch nervous about this - does the license for it allow it to be redistributed?

File: ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Freetype/tasks/main.yml (hunk 4 above)
Comment (sxa555) [reply ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Freetype/tasks/main.yml]: Freetype role being added in https://github.com/AdoptOpenJDK/openjdk-infrastructure/pull/566 superceding this

**Comment by sxa555** [comment]
This would need to be heavily required to be able to merge it now that we have done the work to complete windows machine setup in another change. The certificate stuff isn't in there, but that may be better done via a new PR on top of the latest changes if we're going to include them in the playbooks. Closing for now. If anyone disagrees with this, feel free to reopen :-)
